icu4j/main/classes/core/src/com/ibm/icu/impl/ICUResourceBundle.java

```diff
@@ -421,6 +428,45 @@ public void getAllItemsWithFallback(String path, UResource.Sink sink)
         rb.getAllItemsWithFallback(key, readerValue, sink);
     }
 
+    /**
+     * Locates the resource specified by `path` in this resource bundle (performing any necessary fallback and
+     * following any aliases) and, if the resource is a table resource, iterates over its immediate child resources (again,
+     * following any aliases to get the individual resource values), and calls the specified `sink`'s `put()` method
+     * for each child resource (passing it that resource's key and either its actual value or, if that value is an
+     * alias, the value you get by following the alias).  Then walks back over the bundle's parent chain,
+     * similarly iterating over each parent table resource's child resources.
+     * Does not descend beyond one level of table children.
+     * @param path The path of the desired resource
+     * @param sink A `ResourceSink` that gets called for each child resource of the specified resource (and each child
+     * of the resources in its parent chain).
+     */
+    public void getAllChildrenWithFallback(final String path, final UResource.Sink sink)
+            throws MissingResourceException {
+        class AllChildrenSink extends UResource.Sink {
+            @Override
+            public void put(UResource.Key key, UResource.Value value, boolean noFallback) {
+                UResource.Table itemsTable = value.getTable();
+                for (int i = 0; itemsTable.getKeyAndValue(i, key, value); ++i) {
+                    if (value.getType() == ALIAS) {
+                        // if the current entry in the table is an alias, re-fetch it using getValueWithFallback():
+                        // this will follow the alias (and any aliases it points to) and bring back the real value
+                        String aliasPath = value.getAliasString();
+                        ICUResourceBundle aliasedResource = getAliasedResource(aliasPath, ICUResourceBundle.this.wholeBundle.loader, "", null, 0, new String[] { key.toString() }, null, ICUResourceBundle.this);
```

Comment:
- Long line; please keep it to roughly 100 chars.
- Again, we don't have a container key path. Does it work if you pass in `null` instead of the `new String[] ...` ?
- Over-qualified member access
  - `ICUResourceBundle.this.wholeBundle.loader` --> `wholeBundle.loader`
  - `ICUResourceBundle.this` --> `this`

Reply:
Shortening `ICUResourceBundle.this` to just `this` didn't compile, so I left that alone.  Shortening `ICUResourceBundle.this.wholeBundle.loader` to just `wholeBundle.loader` _did_ work, so I did change that one.

Passing `null` for `baseKeyPath` does seem to work and is parallel with the change I just made on the C++ side, so I'm making the change, although it worries me a little, maybe just because I don't understand this code well enough...